packages/core/core/src/factories/useCartFactory.ts

```diff
@@ -1,27 +1,27 @@
-import { CustomQuery, UseCart } from '../types';
+import { UseCart } from '../types';
 import { Ref, computed } from '@vue/composition-api';
 import { sharedRef } from '../utils';
 
-export type UseCartFactoryParams<CART, CART_ITEM, PRODUCT, COUPON> = {
-  loadCart: (customQuery?: CustomQuery) => Promise<CART>;
+export type UseCartFactoryParams<CART, CART_ITEM, PRODUCT, COUPON, CUSTOM_QUERY = any> = {
```

Comment:
why you made `customQuery` a generic? Do you see any use cases for this? If it's not necessary I'd remove it. cost of removing feature is always higher than adding it so we should avoid features and extension points that are not essential.